httputil/retry_test.go

```diff
@@ -51,6 +52,24 @@ var testRetryStrategy = retry.LimitCount(5, retry.LimitTime(1*time.Second,
 	},
 ))
 
+type counter struct {
+	n  int
```

Comment:
It's not a blocker of course but I've been recently looking at sync/atomic and there are some things that would allow us to increment a counter without a lock.
```
import sync/atomic

return atomic.AddInt64(&cnt.n, 1)
```

Reply:
I know, but for tests  I prefer to go with locks